charting/control.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+import string
+from dataclasses import dataclass, field
+from hashlib import md5
+from typing import Dict, Iterable, List, Union
+from urllib.parse import urlencode
+
+import altair as alt
+import pandas as pd
+from cryptography.fernet import Fernet
+from django.conf import settings
+from pandas.io.formats.style import Styler
+
+
+@dataclass
+class ChartBundle:
+    """
+    The chart bundle contains the dataframe and the altair chart.
+    To create multiple-rendering options.
+
+    """
+
+    label: str
+    alt_title: str
+    df: Union[pd.DataFrame, Styler]
+    chart: alt.Chart
+    logo_src: str = "/static/charting/img/mysociety-logo-white-background.jpg"
+    data_source: str = ""
+    spec: str = field(init=False)
+    ident: str = field(init=False)
+    image_url: str = field(init=False)
+
+    def __post_init__(self):
+        self.spec = self.chart.to_json()
+        self.ident = self.get_ident()
+        self.image_url = self.get_image_url()
+
+    def get_ident(self) -> str:
+        encoded_spec = self.spec.encode("utf-8")
+        hash = md5(encoded_spec).hexdigest()
+        ident = ""
+        for h in hash:
+            if h in string.digits:
+                ident += chr(65 + 8 + int(h))
+            else:
+                ident += h
+
+        return ident[:6]
+
+    def get_image_url(self) -> str:
```

Comment:
as discussed in slack this currently doesn't seem to work so possibly we should remove it and skip the chart if the browser has no JS/ES6 support

Reply:
The alternative @ajparsons and I mentioned in passing earlier this week was to display a `<table>` of the historical sector data by default, and then replace the table with the chart when JS is available. Would that be much work to implement, @ajparsons?